k2/csrc/fsa_equivalent.cc

```diff
@@ -222,6 +231,7 @@ bool IsRandEquivalent(const Fsa &a, const float *a_weights, const Fsa &b,
     CHECK(top_sorted);
     double sum_a =
         ShortestDistance<Type>(a_compose_path, a_compose_weights.data());
+    if (sum_a < best_distance) sum_a = kDoubleNegativeInfinity;
```

Comment:
Here, shouldn't you be doing `continue` instead of sum_a = kDoubleNegativeInfinity ?  Else, I don't see that this function does what it's supposed to do.

Reply:
what if `sum_b >  kDoubleNegativeInfinity`, then we should return false. As `a` has no such a path after pruning, but b has one.